modules/tinymce/src/core/main/ts/content/GetContentImpl.ts

```diff
@@ -16,15 +16,36 @@ const trimEmptyContents = (editor: Editor, html: string): string => {
   return html.replace(emptyRegExp, '');
 };
 
+const bogusAllRegExp = /<(\w+) [^>]*data-mce-bogus="(all|1)"[^>]*>/g;
```

Comment:
@HAFRMO this doesn't look correct... if we have a `all` element then the entire element needs to be stripped not converted into a zero-width space.

I wonder if instead of doing this string based replacement, if instead we should do something like this:
```ts
const parseBody = doc.createElement('body');
parseBody.style.position = 'fixed';
parseBody.style.left = '-9999999px';
parseBody.style.top = '0px';
parseBody.innerHTML = body.innerHTML;

const bogusElements = SelectorFind.decendants(SugarElement.fromDom(parseBody), '[data-mce-bogus]');
Arr.each(bogusElements, (elem) => {
  if (Attribute.get(elem, 'data-mce-bogus') !== 'all) {
    Insert.before(elem, SugarElement.fromText(Unicode.zeroWidth));
  }
  Remove.remove(elem);
});
```

Reply:
Actually the more I think about this, the more I realise that this needs to handle bogus elements in general when getting the text content.

Reply:
Something like this I think should work?
```ts
// Cleanup bogus elements
const bogusElements = SelectorFilter.descendants(SugarElement.fromDom(parseBody), '[data-mce-bogus]');
Arr.each(bogusElements, (elem) => {
  const bogusValue = Attribute.get(elem, 'data-mce-bogus');
  if (bogusValue === 'all') {
    Remove.remove(elem);
  } else if (isBr(elem)) {
    // Need to keep bogus represented as a zero-width space so that they aren't collapsed by the browser
    Insert.before(elem, SugarElement.fromText(Unicode.zeroWidth));
    Remove.remove(elem);
  } else {
    Remove.unwrap(elem);
  }
});
```